src/speechly/client.ts

```diff
@@ -103,11 +102,26 @@ export class Client {
     this.debug = options.debug ?? false
     this.loginUrl = options.loginUrl ?? defaultLoginUrl
     this.appId = options.appId
-
     const apiUrl = generateWsUrl(options.apiUrl ?? defaultApiUrl, language, options.sampleRate ?? DefaultSampleRate)
-    this.apiClient =
-      options.apiClient ??
-      new WebWorkerController(apiUrl)
+    this.apiClient = options.apiClient ?? new WebWorkerController(apiUrl)
+
+    this.storage = options.storage ?? new LocalStorage()
+    this.deviceId = this.storage.getOrSet(deviceIdStorageKey, uuidv4)
+    const storedToken = this.storage.get(authTokenKey)
+
+    // 2. Fetch auth token. It doesn't matter if it's not present.
+    if (storedToken == null || !validateToken(storedToken, this.appId, this.deviceId)) {
+      fetchToken(this.loginUrl, this.appId, this.deviceId).then((token) => {
+        this.authToken = token
```

Comment:
Just as a thought for future: Should choose between stick with "then/catch" and "try/await/catch" we seem to be using both here. Personally, I like the "await" syntax better.